cpp/include/cudf/transform.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cudf.h"
+#include "types.h"
+#include "types.hpp"
+
+namespace cudf {
+
+/**
+ * @brief Creates a new column by applying a unary function against every
+ * element of an input column.
+ *
```

Comment:
Document null behavior, i.e., if `input[i]` is null, `out[i]` will always be null.